x-pack/plugins/apm/public/components/shared/Links/apm/AnomalyDetectionSetupLink.tsx

```diff
@@ -0,0 +1,79 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+import React from 'react';
+import { EuiButtonEmpty, EuiToolTip, EuiIcon } from '@elastic/eui';
+import { i18n } from '@kbn/i18n';
+import { APMLink } from './APMLink';
+import { ENVIRONMENT_NOT_DEFINED } from '../../../../../common/environment_filter_values';
+import { useUrlParams } from '../../../../hooks/useUrlParams';
+import { useFetcher, FETCH_STATUS } from '../../../../hooks/useFetcher';
+
+export function AnomalyDetectionSetupLink() {
+  const { uiFilters, urlParams } = useUrlParams();
+  // check both uiFilters and urlParams for selected environment
+  const environment = uiFilters.environment || urlParams.environment;
```

Comment:
This shouldn't be necessary. If it is we have a bug somewhere. `uiFilters` should read from `urlParams`